NastassiaBon/hometask-3/rainfall.rb

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+ENTER_MESSAGE = 'Enter city name: '
+ERROR_MESSAGE = 'City name can not be blank!'
+EXIT_MESSAGE = 'exit!'
+NUMBER_OF_MONTHS = 12
+
+def data_extraction
+  file = File.new('data.txt')
+  file.read
+end
+
+def mean(town, data)
+  return -1 unless data.include?(town)
+
+  search_town(town, data).sum / NUMBER_OF_MONTHS
+end
+
+def variance(town, data)
+  return -1 unless data.include?(town)
+
+  search_town(town, data).map { |city| (mean(town, data) - city)**2 }.sum / NUMBER_OF_MONTHS
+end
+
+def search_town(town, data)
+  data.split("\n").each do |city|
```

Comment:
Sorry for the previous comment, change city to smth like `line`. Use the name of the entity we iterate over